src/database/DataTypes/ItemStoreJoin.js

```diff
@@ -9,6 +9,7 @@ export class ItemStoreJoin extends Realm.Object {
     // Make the item no longer visible in this store
     const item = itemResults[0];
     item.isVisible = false;
+    item.defaultPrice = 0;
```

Comment:
```suggestion
```
I still think it's not correct. 
1. Line 11 is necessary, being the core purpose of this destructor due to how mobile differs from desktop in handling visibility (it's recorded on item, not ItemStoreJoin).
2. If schema is handling it then it's going into redundant zone.

I'm going to push this along so I can merge it.